File: java-pr/49/java/performance/NumberCalculator07.java
// {fact rule=performance@v1.0 defects=1}
import java.util.*;
import java.util.stream.Collectors;
import java.io.*;

public class NumberCalculator07 {
    private Database database = new Database();
    
    public double calculateSum(List<Double> numbers) {
        double sum = 0;
        for (int i = 0; i < numbers.size(); i++) {
            sum += numbers.get(i);
        }
        return sum;
    }
    
    private static class Database {
        public User findUserById(int id) { return new User(); }
        public List<User> findAllUsers() { return new ArrayList<>(); }
    }
    
    private static class User {}
}
// {/fact}

File: js-pr/56/javascript/code-quality-performance/number_calculator_07.js
// {fact rule=code-quality-performance@v1.0 defects=1}
function calculateSum(numbers) {
    let total = 0;
    for (let i = 0; i < numbers.length; i++) {
        total += numbers[i];
    }
    return total;
}
// {/fact}